Add baseline for activity recognition

as6520 opened the pull request:
In GitLab by @as6520 on Feb 25, 2021, 11:51

This MR updates the protocol to support running multiple algorithms, one of these can be declared baseline.

Depends on https://gitlab.kitware.com/darpa-sail-on/sailon_tinker_launcher/-/merge_requests/3 and https://gitlab.kitware.com/darpa-sail-on/graph-autoencoder/-/merge_requests/16

cfunk1210 commented:
Dismissed my own review since found an issue with adding "detectors" for the there domains.  Since this breaks the other domains, I think we should stage all three here before merging rather than merging a breaking change for some use cases.

as6520 commented:
> Dismissed my own review since found an issue with adding "detectors" for the there domains. Since this breaks the other domains, I think we should stage all three here before merging rather than merging a breaking change for some use cases.

@cfunk1210 can you describe the issue that you are facing?

cfunk1210 commented:
I mentioned this in person today but I was talking about all the other configurations not working since new keywords to the configs have been introduced.  I don't think it's a good idea to push this to master until this has been sorted.

cfunk1210 commented:
do we need to make changes in all adaptors like this for this to work?   Specifically, do we need to change all the adaptors for all domains?  
https://gitlab.kitware.com/darpa-sail-on/graph-autoencoder/-/merge_requests/16/diffs#ed982e77e7ee4c1efa5312d0042e781e5da22fe4_29_29

as6520 commented:
@cfunk1210 I have updated the configs to match with the new convention. To fix the CI tests, we would need to rebase it on #81.

cfunk1210 commented:
Sounds good.  I don't have time to check though

as6520 commented:
Rebasing on #81 to fix the CI pipeline.

as6520 commented:
Depends on #81

as6520 commented:
@nrsyed @waxlamp Could you PTAL at this PR? I have addressed the issues that Chris had raised in the original review. The recent rebase was to fix CI since the sail on the server was updated.

nrsyed reviewed (commented):
Not sure if I've missed anything or if this is even strictly relevant to this PR, but it looks like the structure of the config JSON files has changed so that the `"detector_config"` field is now nested within the `"detectors"` field. Should lines like, for example [this line](https://github.com/darpa-sail-on/sail-on-client/pull/73/files#diff-c54c7bc9fb065bc15831b17931fe45edc31f4b849230c427218a6d5fa9b73cd1R60), use `self.config["detectors"]["detector_config"]` instead of `self.config["detector_config"]`?

as6520 commented:
Good catch @nrsyed! I had not updated CONDDA to match with the detectors convention. I am done with the update, could you take another look?

nrsyed reviewed (approved):
@as6520 LGTM

waxlamp commented on .github/workflows/ci.yml:
```diff
@@ -26,9 +26,10 @@ jobs:
         uses: actions/checkout@v2
         with:
-          repository: darpa-sail-on/sail-on-api
+          repository: darpa-sail-on/Sail-On-API
```
General note: AFAIK, these URLs are not case sensitive. I routinely clone repositories using all lowercase, because I find anything else to be irritating by default.

as6520 replied:
Ah didn't realize that, I'll update it in a separate PR that would address https://github.com/darpa-sail-on/sail-on-client/issues/82

as6520 replied:
@waxlamp would it be okay if I merge this PR?

waxlamp replied:
Yes, that's what I meant when I said to go with Najam's review.

waxlamp reviewed (commented):
I'm not qualified to actually review this PR. However, it looks like Najam has done it, so I would go with his recommendations.
